examples/todo-list/src/models/todo-list.model.ts

```diff
@@ -36,3 +36,8 @@ export class TodoList extends Entity {
     super(data);
   }
 }
+
+export interface TodoListLinks {
+  todos?: (Todo & TodoListLinks)[];
```

Comment:
```ts
todos?: (TodoList & TodoListLinks)[];
```

Reply:
Would it be better to have?
```ts
export type TodoWithLinks = Todo & TodoLinks;
```

So for each model, we can potentially have 3 TypeScript constructs:
- The base model class
- The ModelLinks interface
- The ModelWithLinks type

Reply:
The other option is to have:
- The base ModelType interface
- The ModelLinks interface
- The Model class implements ModelType and ModelLinks and it and its members can be decorated.

Reply:
> Would it be better to have?
> ```ts
> export type TodoWithLinks = Todo & TodoLinks;
> ```
>
> So for each model, we can potentially have 3 TypeScript constructs:
>
> - The base Model class
> - The ModelLinks interface
> - The ModelWithLinks type

Sure 👍 

> The other option is to have:
> (...)
> The Model class implements ModelType and ModelLinks and it and its members can be decorated.

IIUC, that would require model properties to be maintained at two places: in `ModelType` interface and `Model` class. I consider that as not very ergonomic.

Reply:
@raymondfeng  How about calling the new type `TodoListFindResult` instead of `TodoListWithLinks`?

Reply:
> How about calling the new type TodoListFindResult instead of TodoListWithLinks?

Never mind. That would not work well, because then we will end up with `TodoLinks` interface containing a property of type `TodoListFindResult`, which looks weird to me.